tests/serverpod_test_server/test_integration/database_operations/entity_relations/self/one_to_many/count/where_none_test.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:serverpod/database.dart' as db;
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  var session = await IntegrationTestServer().session();
+
+  group('Given entities with one to many relation ', () {
+    tearDown(() async {
+      await Cat.db.deleteWhere(session, where: (_) => db.Constant.bool(true));
+    });
+
+    test(
+      'when counting entities filtered on none many relation then result is as expected.',
+      () async {
+        var zelda = await Cat.db.insertRow(session, Cat(name: 'Zelda'));
+        var smulan = await Cat.db.insertRow(session, Cat(name: 'Smulan'));
+
+        await Cat.db.insert(session, [
+          Cat(name: 'Kitten1', motherId: zelda.id),
+          Cat(name: 'Kitten2', motherId: zelda.id),
+          Cat(name: 'Kitten3', motherId: smulan.id),
+        ]);
+
+        var catCount = await Cat.db.count(
+          session,
+          where: (t) => t.kittens.none(),
+        );
+
+        expect(catCount, 3);
```

Comment:
For my knowledge, this returns the three kittens right?

Reply:
That is correct!